### PR TITLE
AMDGPU: Cleanup selection patterns for buffer loads

### PR DESCRIPTION
**Pull request by arsenm**
We should just support these for all register types.

### CODE REVIEW

**Comment by arsenm** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/95378?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#95379** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/95379?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#95378** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/95378?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#95377** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/95377?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#95376** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/95376?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @arsenm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: llvm/lib/Target/AMDGPU/BUFInstructions.td [hunk 1]
```diff
@@ -1421,27 +1421,21 @@ let OtherPredicates = [HasPackedD16VMem] in {
...
-defm : MUBUF_LoadIntrinsicPat<SIbuffer_load, v4f32, "BUFFER_LOAD_DWORDX4">;
-defm : MUBUF_LoadIntrinsicPat<SIbuffer_load, v4i32, "BUFFER_LOAD_DWORDX4">;
-defm : MUBUF_LoadIntrinsicPat<SIbuffer_load, v2i64, "BUFFER_LOAD_DWORDX4">;
-defm : MUBUF_LoadIntrinsicPat<SIbuffer_load, v2f64, "BUFFER_LOAD_DWORDX4">;
-defm : MUBUF_LoadIntrinsicPat<SIbuffer_load, v8i16, "BUFFER_LOAD_DWORDX4">;
-defm : MUBUF_LoadIntrinsicPat<SIbuffer_load, v8f16, "BUFFER_LOAD_DWORDX4">;
-defm : MUBUF_LoadIntrinsicPat<SIbuffer_load, v8bf16, "BUFFER_LOAD_DWORDX4">;
+foreach vt = Reg32Types.types in {
+defm : MUBUF_LoadIntrinsicPat<SIbuffer_load, vt, "BUFFER_LOAD_DWORD">;
+}
```
Comment (jayfoad) [inline llvm/lib/Target/AMDGPU/BUFInstructions.td:1426]: Don't need the braces.

**Review by jayfoad — APPROVED** [review]
Nice!

File: llvm/lib/Target/AMDGPU/BUFInstructions.td (hunk 1 above)
Comment (arsenm) [reply llvm/lib/Target/AMDGPU/BUFInstructions.td:1426]: I'm not a big fan of omitting the braces, especially in tablegen. If we're going to delete the braces the lines should at least be indented

**Review by Sisyph — APPROVED** [review]
LGTM

**Pull request merged**